src/writer/tagvalue/tagvalue_writer.py

```diff
@@ -0,0 +1,87 @@
+#  Copyright (c) 2022 spdx contributors
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#    http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import TextIO
+
+from src.model.document import Document
+from src.writer.tagvalue.annotation_writer import write_annotation
+from src.writer.tagvalue.creation_info_writer import write_creation_info
+from src.writer.tagvalue.extracted_licensing_info_writer import write_extracted_licensing_info
+from src.writer.tagvalue.file_writer import write_file
+from src.writer.tagvalue.package_writer import write_package
+from src.writer.tagvalue.relationship_writer import write_relationship
+from src.writer.tagvalue.snippet_writer import write_snippet
+from src.writer.tagvalue.tagvalue_writer_helper_functions import write_separator, scan_relationships, \
+    determine_files_containing_snippets, write_optional_heading, write_list_of_elements
+
+
+def write_document_to_file(document: Document, file_name: str):
+    with open(file_name, "w") as out:
+        write_document(document, out)
+
+
+def write_document(document: Document, text_output: TextIO):
+    """
+    Write a SPDX tag value document.
+    - document - src.document instance.
+    - text_output - file like object that will be written to.
+    """
+
+    text_output.write("## Document Information\n")
+    # Write out creation info
+    write_creation_info(document.creation_info, text_output)
+    write_separator(text_output)
+
+    # Write sorted annotations
+    write_optional_heading(document.annotations, "## Annotations\n", text_output)
+    write_list_of_elements(document.annotations, write_annotation, text_output, True)
```

Comment:
According to [spec](https://spdx.github.io/spdx-spec/v2.3-RC1/composition-of-an-SPDX-document/), annotations are the last thing to be written in an SPDX document.

On that note: We should change change the ordering of the values in the individual writers to reflect the ordering of the specification.